trin-core/src/portalnet/types/metric.rs

```diff
@@ -45,6 +96,41 @@ mod test {
         }
     }
 
+    #[test]
+    fn distance_log2() {
+        fn prop(x: DhtPoint) -> TestResult {
+            let x = U256::from_big_endian(&x.0);
+            let distance = Distance(x);
+            let log2_distance = distance.log2();
+
+            match log2_distance {
+                Some(log2) => {
+                    let x_floor = U256::from(1u8) << log2 - 1;
```

Comment:
I'm a fan of being super explicit with order of operations. So as a reader I find it quicker to confidently comprehend:
```suggestion
                    let x_floor = U256::from(1u8) << (log2 - 1);
```